.github/actions/workflow-run-job-linux/action.yml

```diff
@@ -0,0 +1,197 @@
+# using the host's docker daemon.
+#
+# This allows us to run actions that require node v20 in the `cruizba/ubuntu-dind:jammy-26.1.3` container, and
+# then launch our Ubuntu18.04-based GCC 6/7 containers to build and test CCCL.
+#
+# The main inconvenience to this approach is that any container mounts have to match the paths of the runner host,
+# not the paths as seen in the intermediate (`cruizba/ubuntu-dind`) container.
+#
+# Note: I am using `cruizba/ubuntu-dind:jammy-26.1.3` instead of `docker:latest`, because GitHub doesn't support
+# JS actions in alpine aarch64 containers, instead failing actions with this error:
+# ```
+# Error: JavaScript Actions in Alpine containers are only supported on x64 Linux runners. Detected Linux Arm64
+# ```
+
+inputs:
+  id:
+    description: "A unique identifier."
+    required: true
+  command:
+    description: "The command to run."
+    required: true
+  image:
+    description: "The Docker image to use."
+    required: true
+  runner:
+    description: "The GHA runs-on value."
+    required: true
+  cuda:
+    description: "The CUDA version to use when selecting a devcontainer."
+    required: true
+  host:
+    description: "The host compiler to use when selecting a devcontainer."
+    required: true
+
+runs:
+  using: "composite"
+  steps:
+    - name: Install dependencies
+      shell: sh
+      run: |
+        # Install script dependencies
+        apt update
+        apt install -y --no-install-recommends tree git
+    - name: Checkout repo
+      uses: actions/checkout@v4
+      with:
+        path: ${{github.event.repository.name}}
+        persist-credentials: false
+    - name: Add NVCC problem matcher
+      shell: bash --noprofile --norc -euo pipefail {0}
+      run: |
+        echo "::add-matcher::${{github.event.repository.name}}/.github/problem-matchers/problem-matcher.json"
+    - name: Get AWS credentials for sccache bucket
+      uses: aws-actions/configure-aws-credentials@v4
+      with:
+        role-to-assume: arn:aws:iam::279114543810:role/gha-oidc-NVIDIA
+        aws-region: us-east-2
+        role-duration-seconds: 43200 # 12 hours
+    - name: Run command # Do not change this step's name, it is checked in parse-job-times.py
+      shell: bash --noprofile --norc -euo pipefail {0}
+      env:
+        CI: true
+        RUNNER: "${{inputs.runner}}"
+        # Dereferencing the command from an env var instead of a GHA input avoids issues with escaping
+        # semicolons and other special characters (e.g. `-arch "60;70;80"`).
+        COMMAND: "${{inputs.command}}"
+        AWS_ACCESS_KEY_ID: "${{env.AWS_ACCESS_KEY_ID}}"
+        AWS_SESSION_TOKEN: "${{env.AWS_SESSION_TOKEN}}"
+        AWS_SECRET_ACCESS_KEY: "${{env.AWS_SECRET_ACCESS_KEY}}"
+      run: |
+        echo "[host]      github.workspace: ${{github.workspace}}"
+        echo "[container] GITHUB_WORKSPACE: ${GITHUB_WORKSPACE:-}"
+        echo "[container]              PWD: $(pwd)"
+
```

Comment:
Feel free to delete these. I had them in there to validate the host/container paths for docker-outside-of-docker:

```suggestion
```